### ISSUE TITLE
Resuming training from saved checkpoint produces different result than uninterrupted training

### ISSUE THREAD

**Issue by lostmsu** [tensorflow/tensorflow#27049]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes ([Google Collab](https://colab.research.google.com/drive/1ZRhX4VBEWo4fh4zbXL_p-HDCUGTRelnH))
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Google Collab
- TensorFlow installed from (source or binary): Google Collab
- TensorFlow version (use command below): 1.13.1
- Python version: Python 3

**Describe the current behavior**
Loading a model with `tf.keras.models.load_model`, produced with `tf.keras.callbacks.ModelCheckpoint` , and resuming training produces different results from running the training without save model + restore model in the middle.

**Describe the expected behavior**
Saving and restoring the model should allow to resume training as if there was no interruption in the first place.

**Code to reproduce the issue**
[Google Collab](https://colab.research.google.com/drive/1ZRhX4VBEWo4fh4zbXL_p-HDCUGTRelnH)

**Other info / logs**
No interruption:

> Epoch 49/100
>  - 0s - loss: 3.5190 - val_loss: 3.3597
> Epoch 50/100
>  - 0s - loss: 3.4090 - val_loss: 3.2668
> Epoch 51/100
>  - 0s - loss: 3.2637 - val_loss: 3.1623
> Epoch 52/100
>  - 0s - loss: 3.0962 - val_loss: 2.9975

With interruption:

> Epoch 49/50
>  - 0s - loss: 3.5190 - val_loss: 3.3597
> Epoch 50/50
> 
> Epoch 00050: saving model to weights.50.ckpt
>  - 0s - loss: 3.4090 - val_loss: 3.2668
> ... `load_model('weights.50.ckpt')` ...
> Epoch 51/100
>  - 0s - loss: 3.2637 - val_loss: 3.3816
> Epoch 52/100
>  - 0s - loss: 3.3175 - val_loss: 3.1457

The model does not have any random elements, so it looks like optimizer state is lost.

**Comment by allenlavoie** [comment]
Model.fit is taking random slices of the data and batching them together. If you control for that, e.g. make all of the examples identical, can you still reproduce?

**Comment by lostmsu** [comment]
@allenlavoie the model will converge instantly if there's just 1 training sample.
I added `shuffle=False` to `.fit(...)` calls. The issue persisted.

**Comment by allenlavoie** [comment]
Hrm. It looks like it's passing include_optimizer to Model.save. @tanzhenyu, any ideas?

**Comment by tanzhenyu** [comment]
@lostmsu The previous keras optimizer is problematic. We have provided a new set of optimizers under exactly the same name and fully backward compatible for users. However as unfortunate as this is, the new ones did not make to tf 1.13.1. If you do !pip install tf-nightly-gpu-2.0-preview, then you should be able to see identical behavior.

That said, you need to change one line from tf.set_random_seed(seed) to tf.random.set_seed(seed)

**Comment by lostmsu** [comment]
@tanzhenyu confirming, seems to be fixed in 2.0-preview
